fv3gfs-util/fv3gfs/util/partitioner.py

```diff
@@ -812,17 +843,33 @@ def subtile_slice(
         global_extent: size of the tile or cube's computational domain
         layout: the (y, x) number of ranks along each tile axis
         subtile_index: the (y, x) position of the rank on the tile
+        edge_tile_ratio: ratio between interior and boundary tile sizes.
+            Assumes full integer divisibility for both interior 
+            and boundary tile sizes.
         overlap: whether to assign regions which belong to multiple ranks
             to both ranks, or only to the higher rank (default)
     """
     return_list = []
     # discard last index for interface variables, unless you're the last rank
     # done so that only one rank is responsible for the shared interface point
-    for index in _index_generator(dims, global_extent, subtile_index, layout):
-        start = index.i_subtile * index.base_extent
+    for index in _index_generator(dims, global_extent,
+                                  subtile_index, layout, edge_tile_ratio):
+        if index.i_subtile == 0 or index.n_ranks < 3:
+            start = index.i_subtile * index.base_extent
+        elif index.is_end_index:
+            start = normal_round((1 + (index.i_subtile - 1) / index.edge_tile_ratio) * index.base_extent)
+        else:
+            start = normal_round((index.i_subtile - 1 + index.edge_tile_ratio) * index.base_extent)
         if index.is_end_index or overlap:
             end = start + index.extent
         else:
             end = start + index.base_extent
-        return_list.append(slice(start, end))
+        return_list.append(slice(int(start), int(end)))
     return tuple(return_list)
+
+
+def normal_round(n):
+    "Needed to get around IEEE754 half-to-even rounding. Will not work with negative numbers."
```

Comment:
Why is it bad to have half-to-even rounding here? As a user, I'd probably expect rounding to occur with the behavior of Python's default `round`.

Reply:
Behaviour changed to use native python rounding. I found avoiding half-to-even-rounding gave better results, but with the new approach this doesn't matter anymore.